piksi_tools/bootload.py

```diff
@@ -81,11 +81,16 @@ def wait_for_handshake(self, timeout=None):
     if timeout is not None:
       t0 = time.time()
     self.handshake_received = False
+    expire = time.time() + 15.0
+    self.link.send(SBP_MSG_RESET, "")
     while not self.handshake_received:
       time.sleep(0.1)
       if timeout is not None:
         if time.time()-timeout > t0:
           return False
+      if time.time() > expire:
+        expire = time.time() + 15.0
+        self.link.send(SBP_MSG_RESET, "")
```

Comment:
Yeah I think we should roll `reply_handshake` into it and have one function that takes the device from whatever state it's currently in and puts it in the bootloader, ready to flash. I think `handshake` would be good - do you think that its obvious that would take you from the application firmware into the bootloader?